service/devdoc/twin_requirements.md

```diff
@@ -103,28 +103,44 @@ having to get it first which can be useful in some high performance scenarios.
 ### get(done)
 The `get` method is used to refresh the `Twin` instance with the latest values from the IoT Hub Registry.
 
-**SRS_NODE_IOTHUB_TWIN_16_020: [** The `get` method shall call the `getTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
+**SRS_NODE_IOTHUB_TWIN_16_020: [** If `this.moduleId` is `null`, The `get` method shall call the `getTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
 - `this.deviceId`
 - `done`
 **]**
 
+**SRS_NODE_IOTHUB_TWIN_18_001: [** If `this.moduleId` is not `null`, the `get` method shall call the `getModuleTwin` method of the `Registry` instance stored in `_registry` property with the following parameters:
+  - `this.deviceId`
+  - `this.moduleId`
+  - `done`
+**]**
+
 ### update(patch, done)
-The `update` method is used to update any part of the device twin stored in the IoT Hub registry with the values passed as a patch argument. 
+The `update` method is used to update any part of the device twin stored in the IoT Hub registry with the values passed as a patch argument.
 To satisfy optimistic concurrency requirements, the HTTP request `If-Match` header will be populated with the `Twin.eTag` field if defined. If not defined, `*` will be used.
```

Comment:
>If not defined, `*` will be used [](start = 140, length = 32)

In reference to another comment where I asked about what happens if an etag is not present, (configuration registry requirements), here is a requirement that deals with that case.

Reply:
Yep, very similar.  The module and configuration APIs have a "forceUpdate" which is required if etag is falsy.  I'm not sure which style is better this (implied forceUpdate) or that (explicit forceUpdate).  My gut says explicit.

---
In reply to: [176865034](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176865034) [](ancestors = 176865034)